doc/adrs/086_allow_single_click_tools_to_be_executed_on_edges.adoc

```diff
@@ -0,0 +1,99 @@
+= ADR-086 - Allow single click tools to be executed on edges
+
+== Context
+
+Today a single click tool can only be executed on Nodes. An enhancement could be to allow single click tools to also be executed on Edges as it is the case in Sirius Desktop.
+
+== Decision
+
+Single click tools can also be executed on Edges.
+
+=== Frontend
+
+In `DiagramRepresentation.types.ts`, a new interface named `DiagramElementDescription` will be introduced.
+`NodeDescription` (there is no `EdgeDescription` in this file) will implement it.
+The target descriptions of a `SingleClickOnDiagramElementTool` will be `DiagramElementDescription` instead of `NodeDescription`.
+
+[source,ts]
+----
+export interface DiagramElementDescription {
+  id: string;
+}
+
+export interface NodeDescription extends DiagramElementDescription {
+  id: string;
+}
+
+export interface SingleClickOnDiagramElementTool extends Tool {
+  appliesToDiagramRoot: boolean;
+  selectionDescriptionId: string;
+  targetDescriptions: DiagramElementDescription[];
+}
+----
+
+=== Backend
+
+==== sirius-components-collaborative-diagrams
+
+In `diagram.graphqls`, a new interface `DiagramElementDescription` will be introduced. `NodeDescription` and `EdgeDescription` will implement it.
+
+The target descriptions of a `SingleClickOnDiagramElementTool` will be `DiagramElementDescription` instead of `NodeDescription`.
+
+[source,graphql]
+----
+interface DiagramElementDescription {
+  id: ID!
+  synchronizationPolicy: SynchronizationPolicy!
+}
+
+type NodeDescription implements DiagramElementDescription {
+  id: ID!
+  synchronizationPolicy: SynchronizationPolicy!
+  borderNodeDescriptions: [NodeDescription!]!
+  childNodeDescriptions: [NodeDescription!]!
+}
+
+type EdgeDescription implements DiagramElementDescription {
+  id: ID!
+  synchronizationPolicy: SynchronizationPolicy!
+  sourceNodeDescriptions: [NodeDescription!]!
+  targetNodeDescriptions: [NodeDescription!]!
+}
+
+type SingleClickOnDiagramElementTool implements Tool {
+  id: ID!
+  label: String!
+  imageURL: String!
+  appliesToDiagramRoot: Boolean!
+  selectionDescriptionId: String
+  targetDescriptions: [DiagramElementDescription!]!
+}
+----
+
+In `InvokeSingleClickOnDiagramElementToolEventHandler.java`, the `executeTool` method shall handle edges in addition to nodes.
+
+==== sirius-components-diagrams
+
+A new interface named `IDiagramElementDescription.java` will be introduced.
+`NodeDescription.java` and `EdgeDescription.java` will implement it.
+
+[source,java]
+----
+public interface IDiagramElementDescription {
+    UUID getId();
+}
+----
+
+The target descriptions of `SingleClickOnDiagramElementTool.java` will be `DiagramElementDescription` instead of `NodeDescription`.
+
+==== sirius-components-view-emf
+
+`ViewDiagramDescriptionConverter.java` && `ViewToolSectionsProvider.java` will be updated to handle edges in addition to nodes.
```

Comment:
A single '&' should be enough in prose :)

Reply:
I don't think these changes are enough to be able to use this new feature in View-based modelers. There is no way in the DSL itself to make `ViewDiagramDescriptionConverter` produce a `SingleClickOnDiagramElementTool` that targets an `EdgeDescription`.

There's an overlap with the upcoming work on providing more control in the View DSL on which tools are availble on the palette of which element, so I don't think it's worth trying to fix that in this PR, but the changes `sirius-components-view-emf` can probably be dropped (and the ADR changed to mention this is out of scope for the moment).

Reply:
I would not automatically drop this part of the work if it's halfway there. We will have very soon (today probably) five PRs which overlap one another on tools so one more is not really an issue.

Reply:
Yes, today, in `ViewDiagramDescriptionConverter` &  `ViewToolSectionsProvider`, only tools applied on diagram background are properly handled. So, I've only updated code to handle the new `IDiagramElementDescription ` interface. I will also add an out-of-scope mention to the sirius-components-view-emf section.